File: BeitPatissierServer/Models/RecipeAllergen.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace BeitPatissierServer.Models
{
    public class RecipeAllergen
    {
        public int RecipeId { get; set; }
        [ForeignKey(nameof(RecipeId))]
        public Recipe Recipe { get; set; }

        public int AllergenId { get; set; }
        [ForeignKey(nameof(AllergenId))]
        public Allergen Allergen { get; set; }
    }
}

File: beit-patissier-manage/src/models/RecipeAllergen.ts
import { Recipe } from './Recipe';
import { Allergen } from './Allergen';

export class RecipeAllergen {
    recipeId: number;
    recipe: Recipe;
    allergenId: number;
    allergen: Allergen;

    constructor(
        recipeId: number,
        recipe: Recipe,
        allergenId: number,
        allergen: Allergen
    ) {
        this.recipeId = recipeId;
        this.recipe = recipe;
        this.allergenId = allergenId;
        this.allergen = allergen;
    }
}
